### ISSUE TITLE
Xcode 9 beta 5 compilation error

### ISSUE THREAD

**Issue by mecid** [ReactiveKit/Bond#449]
/Pods/Bond/Sources/Bond/Shared/NSObject+KVO.swift:331:47: Value of optional type 'Selector?' not unwrapped; did you mean to use '!' or '?'?

**Comment by mecid** [comment]
@srdanrasic any idea why this error appear?

**Comment by srdanrasic** [comment]
Are you on swift-4 branch?

**Comment by mecid** [comment]
@srdanrasic sure)

**Comment by srdanrasic** [comment]
It's very strange that you are getting this error as I have no problem compiling swift-4 branch with Xcode 9b5. Also, looking at the code where the selector is defined - the method does not return an optional:

```swift
func sel_registerName(_ str: UnsafePointer<Int8>) -> Selector
```

This feels more like CocoaPods integration issues...

**Comment by sathishkraj** [comment]
@srdanrasic is swift-4 branch swift 3.2 compatible?

**Comment by mecid** [comment]
@srdanrasic swift-4 branch have problems in Diff which not for swift-4.

**Comment by srdanrasic** [comment]
Are you using CocoaPods? `swift-4` branch only works with Carthage or manual integration because I don't know of a way to tell CocoaPods to use specific branch for a dependency.

**Comment by mecid** [comment]
@srdanrasic we are using cocoapods.
You can do it like this:
`pod 'Bond', :git => 'https://github.com/ReactiveKit/Bond.git', :branch => 'swift-4'`

**Comment by srdanrasic** [comment]
The problem is when specifying a dependency to another pod, you can't use branch.

https://github.com/ReactiveKit/Bond/blob/master/Bond.podspec#L33

I'm not sure if there is a workaround around that.

**Comment by mecid** [comment]
@srdanrasic so, we can't use Bond with Swift-4?

**Comment by srdanrasic** [comment]
Unfortunately not with CocoaPods at the moment. It's a limitation of CocoaPods. The problem will be solved by the time Xcode 9 is out of beta.

**Comment by codedbypm** [comment]
Having the same issue with XCode 9 beta 6 and Cocoapods. 

> The problem will be solved by the time Xcode 9 is out of beta.

If the issue is cocoapods related, how certain is your statement?

**Comment by codedbypm** [comment]
What about as a temporary fix a guard in `_swizzleDeinit(:)`? 
```
guard let selector = sel_registerName("dealloc") else {
    return
}
```
Isn't safe to assume that the call to `sel_registerName` will never return an optional?

**Comment by srdanrasic** [comment]
I'm positive. The issue is related to CocoaPods. Until Swift 4 is ready, and both ReactiveKit and Diff.swift (Bond dependencies) are out with Swift 4 version, Bond for Xcode 9 will not be available through CocoaPods. Hopefully, we will be able to solve this within a few days after Xcode 9 is out from beta.

**Comment by marclove** [comment]
@srdanrasic does this mean Bond won't be supported on Xcode 9 with Swift 3.2? My project is stuck at 3.2 for reasons outside my control and Xcode 9 is becoming necessary due to iPhone X.

**Comment by srdanrasic** [comment]
I'm pleased to announce that Xcode 9 supported now - just update to 6.3.0!

You can use the framework both from Swift 3.2 and Swift 4 code. 

If you encounter any new issues, please open a new ticket. Check out this note if you are using observable collections: https://github.com/ReactiveKit/Bond#note-xcode-9-support

Thank you all!

**Comment by mecid** [comment]
@srdanrasic i have issue with Bond dependency Diff.
<img width="1434" alt="screen shot 2017-09-13 at 16 09 56" src="https://user-images.githubusercontent.com/324372/30376604-086eb978-989e-11e7-9336-30ed6fcde9a2.png">

**Comment by srdanrasic** [comment]
Ah, damn, seems like CocoaPods is still not happy. Can you this this:
https://github.com/CocoaPods/CocoaPods/issues/6791#issuecomment-320028044

**Comment by srdanrasic** [comment]
More info about CocoaPods integration here: https://github.com/ReactiveKit/Bond/issues/460#issuecomment-329277714

**Comment by mecid** [comment]
thanks

**Comment by joshtobin** [comment]
I think this has been broken again in 6.4.0

**Comment by srdanrasic** [comment]
@joshtobin If you have added a script to your Podspec as described in #460, you should remove that now.

**Comment by joshtobin** [comment]
@srdanrasic no I hadn't used that script, this was actually just while I was updating to Xcode9, one developer on the team updated pods to 6.3.0 and it was working fine, I updated and got v6.4.0 and it didn't work. I have fixed the pod version to 6.3.0 and it's working again now, so I honestly think 6.4.0 is causing it again.
